src/Http/Routing/test/UnitTests/Builder/MapEndpointEndpointDataSourceBuilderExtensionsTest.cs

```diff
@@ -153,10 +155,57 @@ public void MapEndpoint_GeneratedDelegateWorks()
             Assert.IsType<Metadata>(endpointBuilder1.Metadata[0]);
         }
 
+        [Fact]
+        public void MapEndpoint_HttpMethodMetadataPrecedence()
```

Comment:
This is porbably worth a comment since the intent isn't clear from the test name.